src/Api/Data/PaginatedDatasourceApiResult.php

```diff
@@ -0,0 +1,17 @@
+<?php declare(strict_types=1);
+
+namespace Torr\Storyblok\Api\Data;
+
+use Torr\Storyblok\Datasource\DatasourceEntry;
+
+final class PaginatedDatasourceApiResult
+{
+	/**
+	 * @param DatasourceEntry[] $entries
+	 */
+	public function __construct (
+		public readonly int $perPage,
+		public readonly int $totalPages,
+		public readonly array $entries,
+	) {}
```

Comment:
Let's combine `PaginatedDatasourceApiResult` & `PaginatedStoryApiResult`

Reply:
Do you want a generic `PaginatedApiResult` with a PHPStan generic definition to provide a type to the inner entries?

Reply:
Yes 👍